PKHeX/Legality/Checks.cs

```diff
@@ -782,7 +803,7 @@ private CheckResult verifyEncounterG12()
                 pkm.WasEgg = true;
                 return verifyEncounterEgg();
             }
-            EncounterMatch = EncounterOriginalGB = EncountersGBMatch.FirstOrDefault()?.Encounter;
+            EncounterMatch = EncounterOriginalGB = EncountersGBMatch?.FirstOrDefault()?.Encounter;
```

Comment:
EncountersGBMatch is already checked for null at the top of the method; null check is unnecessary at this point as nothing changes what EncountersGBMatch is